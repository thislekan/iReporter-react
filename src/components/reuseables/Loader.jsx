import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
import { BounceLoader } from 'react-spinners';
import style from '../../styles/loader.css';

const override = css`
    display: block;
    margin: 0 auto;
    border-color: red;
`;

function Loader(props) {
  const { isLoading } = props;
  return (
    <div className={style.loader_div}>
      <BounceLoader
        css={override}
        sizeUnit="px"
        size={200}
        loading={isLoading}
        color={'#117d5a'}
      />
    </div>
  );
}

Loader.propTypes = {
  isLoading: PropTypes.bool,
};

Loader.defaultProps = {
  isLoading: false,
};

export default Loader;
