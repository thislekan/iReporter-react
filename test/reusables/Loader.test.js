import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../src/components/reuseables/Loader.jsx';

describe('<Loader />', () => {
  it('should render the PageLoader', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
