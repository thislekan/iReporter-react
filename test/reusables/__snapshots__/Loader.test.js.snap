// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the PageLoader 1`] = `
<div>
  <onlyUpdateForKeys(Loader)
    color="#117d5a"
    css={
      Object {
        "map": undefined,
        "name": "1lkjh1n",
        "next": undefined,
        "styles": "
    display: block;
    margin: 0 auto;
    border-color: red;
",
      }
    }
    loading={false}
    size={200}
    sizeUnit="px"
  />
</div>
`;
